WooCommerce/Classes/ViewRelated/Orders/Order Creation/NotesSection/EditOrderNotesViewModel.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import Combine
+
+class OrderCreationNotesViewModel: EditCustomerNoteViewModelProtocol {
+    /// Store for the edited content
+    ///
+    @Published var newNote: String
+
+    /// Defines the navigation right button state
+    ///
+    @Published private(set) var navigationTrailingItem: EditCustomerNoteNavigationItem = .done(enabled: false)
+
+    /// Not used.
+    ///
+    @Published var presentNotice: EditCustomerNoteNotice? = nil
+
+    /// Not used.
+    ///
+    var presentNoticePublisher: Published<EditCustomerNoteNotice?>.Publisher {
+        $presentNotice
+    }
+
+    /// Commit the original note.
+    ///
+    func updateNote(onFinish: @escaping (Bool) -> Void) {
+        originalNote = newNote
+        onFinish(true)
+
+        updateNoteAnalyticsTrackAction()
+    }
+
+    /// Revert to original content.
+    ///
+    func userDidCancelFlow() {
+        newNote = originalNote
+    }
+
+    /// Stores the original note content.
+    ///
+    @Published private var originalNote: String
+
+    /// Analytics engine.
+    ///
+    private let updateNoteAnalyticsTrackAction: () -> Void
```

Comment:
I don't think this is needed! Since `newNote` is a published property, anyone can subscribe to it and perform any task, like updating analytics.

Reply:
Indeed! I've removed this and went with a different strategy for the analytics trigger following how the `NewOrderViewModel` handles analytics. Let me know what you think about it.